### PR TITLE
Feat/otlp functional design plat 535

### PR DESCRIPTION
**Pull request by gvelez17**
## Description

Switch from classes to flatter more functional design

switch from prometheus-exporter to otlp-http generic exporter for metrics

Include relevant motivation, context, brief description and impact of the change(s). List follow-up tasks here.

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [ ] Test A (e.g. Test A - New test that ... ran in local, docker, and dev unstable.)
- [ ] Test B

## PR checklist

Before submitting this PR, please make sure:

- [ ] I have tagged the relevant reviewers and interested parties
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

File: packages/common/src/logger-base.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class ServiceLoggerBase {
     message = `[${now.toUTCString()}] service=${this.service} ${message}`
 
     if (this.logToConsole) {
-      console.log(message)
+      //console.log(message)
```
Comment (oed) [inline packages/common/src/logger-base.ts:121]: Why is this commented out?

File: packages/http-client/src/__tests__/ceramic-http-client.test.ts [hunk 2]
```diff
@@ -37,6 +37,10 @@ const second = {
 const streamId = new StreamID(0, FAKE_CID_1)
 
 describe('applyCommit', () => {
+  afterEach(() => {
+    delete process.env.OTEL_EXPORTER_PROMETHEUS_HOST;
+    delete process.env.OTEL_EXPORTER_PROMETHEUS_PORT;
```
Comment (oed) [inline packages/http-client/src/__tests__/ceramic-http-client.test.ts:42]: Why are these here?

File: package.json [hunk 3]
```diff
@@ -37,6 +37,13 @@
   },
   "prettier": "eslint-config-3box/prettier.config",
   "dependencies": {
+    "@opentelemetry/api": "^1.1.0",
+    "@opentelemetry/api-metrics": "^0.28.0",
+    "@opentelemetry/core": "^1.2.0",
+    "@opentelemetry/exporter-metrics-otlp-http": "^0.28.0",
```
Comment (oed) [inline package.json:43]: Not used

File: package.json [hunk 4]
```diff
@@ -37,6 +37,13 @@
...
+    "@opentelemetry/exporter-prometheus": "^0.28.0",
+    "@opentelemetry/sdk-metrics-base": "^0.28.0",
```
Comment (oed) [inline package.json:45]: These should be in the `common` package.

File: packages/common/src/metrics-setup.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+
+import { MeterProvider } from '@opentelemetry/sdk-metrics-base'
+import { PrometheusExporter } from '@opentelemetry/exporter-prometheus'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
```
Comment (oed) [inline packages/common/src/metrics-setup.ts:4]: This dependency is missing.

**Review by oed — CHANGES_REQUESTED** [review]

File: packages/http-client/src/__tests__/ceramic-http-client.test.ts (hunk 2 above)
Comment (gvelez17) [reply packages/http-client/src/__tests__/ceramic-http-client.test.ts:42]: This was an earlier attempt to deal with EADDR_IN_USE test failures, these were actually because of parallelization

File: package.json (hunk 3 above)
Comment (gvelez17) [reply package.json:43]: hm I thought I had run `npm uninstall -save` but maybe I had not - removed!

File: package.json [hunk 6]
```diff
@@ -37,6 +37,13 @@
...
+    "@opentelemetry/sdk-node": "^0.28.0",
```
[the remark this answers is not in the document]
Comment (gvelez17) [reply package.json:46]: hm exporter-prometheus is definitely used, how are we seeing this is unused?

File: package.json (hunk 4 above)
Comment (gvelez17) [reply package.json:45]: got it, moving

File: package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (oed) [reply package.json]: Mainly pointing to `@opentelemetry/sdk-node` here.

**Comment by gvelez17** [comment]
got messy commit history with a rebase, closing and moving the changes to https://github.com/ceramicnetwork/js-ceramic/pull/2195
